### PR TITLE
[language][diem-framework] rewrite account_limits/basics.move as a unit test

### PR DESCRIPTION
**Pull request by vgao1996**
This rewrites `move-lang/functional-tests/tests/diem/account_limits/basics.move` as a Move unit test in diem-framework.

### CODE REVIEW

File: diem-move/diem-framework/core/tests/AccountLimits.move [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            100, /* new_max_inflow */
+            200, /* new_max_outflow */
+            150, /* new_max_holding_balance */
+            10000, /* new_time_period */
+        )
+    }
+
+    #[test(dr = @DiemRoot, tc = @TreasuryCompliance, bob = @0x2)]
+    #[expected_failure(abort_code = 5)]
+    fun update_limits_definition_non_exsistent(dr: signer, tc: signer, bob: signer) {
```
Comment (bob-wilson) [inline diem-move/diem-framework/core/tests/AccountLimits.move:128]: typo in this function name: "exsistent"

Comment (bob-wilson) [reply diem-move/diem-framework/core/tests/AccountLimits.move:128]: I see the same typo in a few other places in this file, so be sure to fix them all

**Review by bob-wilson — COMMENTED** [review]
Despite it being my own name, I'm not a big fan of using "bob" to name accounts. Since the setup function is setting up "bob" with `new_parent_vasp_role`, would it make sense to use "vasp" instead?

I like the way you've given the test functions meaningful names. It would be even better IMO to add comments to explain what each test is doing and reasons for expected failures.

**Comment by vgao1996** [comment]
@bob-wilson Sounds reasonable to me. Let me change it to "vasp" (or whatever else that's appropriate and indicative.) @tzakian Can you confirm that the operations here (update limits, publish window) are indeed specific to parent vasps? Thanks!

> It would be even better IMO to add comments to explain what each test is doing and reasons for expected failures.

Will do.

**Comment by tzakian** [comment]
@vgao1996 Correct! These operations are specific to parent VASPs (since it limits the whole VASP in terms of in/outflow of money)

**Comment by vgao1996** [comment]
Closing since this has been cherry-picked onto https://github.com/diem/diem/pull/10058

**Pull request closed without merging**